packages/sources/alongside-test/src/endpoint/balances.ts

```diff
@@ -0,0 +1,114 @@
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+import { customSettings } from '../config'
+import CryptoJS from 'crypto-js'
+import { makeLogger } from '@chainlink/external-adapter-framework/util/logger'
+
+const logger = makeLogger('Alongside Balances Logger')
+
+export const inputParameters = {
+  type: {
+    required: true,
+    description: 'The type of balance',
+    type: 'string',
+  },
+} as const
+
+export type RequestParams = {
+  type: string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: RequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: never
+    ResponseBody: ProviderResponseBody
+  }
+}
+
+export interface ProviderResponseBody {
+  balances: [
+    {
+      symbol: string
+      amount: string
+      holds: string
+      bonded_amount: string
+      reserved_amount: string
+      unbonding_amount: string
+      unvested_amount: string
+      pending_rewards_amount: string
+      past_rewards_amount: string
+      bondable_amount: string
+      withdrawable_amount: string
+    },
+  ]
+  type: string
+  trading_balances: {
+    total: string
+    holds: string
+  }
+  vault_balances: {
+    total: string
+    holds: string
+  }
+}
+
+const sign = (str: string, secret: string) => {
+  const hash = CryptoJS.HmacSHA256(str, secret)
+  return hash.toString(CryptoJS.enc.Base64)
+}
+
+export const httpTransport = new HttpTransport<EndpointTypes>({
```

Comment:
Checking with the person who requested this but if this EA is only used once a day, we may need to change the HTTP transport implementation to use a `foregroundExecute`. The subscription would expire by the time the next request came in anyways

Reply:
Just confirmed that it's currently set for 3x/day so we should move to using a foreground execute